crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -191,24 +198,51 @@ impl BaseClient {
                     #[rustfmt::skip]
                     warn!(
                         ?room_id, event_id,
-                        "Failed to deserialize ephemeral room event: {e}"
+                        "Failed to deserialize read receipt room event: {e}"
                     );
                 }
             }
 
-            // Also include the receipts in the room update, so the timeline is aware of
-            // those. We assume that those happen only in joined rooms.
-            let room_update =
-                new_rooms.join.entry(room_id.clone()).or_insert_with(JoinedRoom::default);
-            room_update.ephemeral.push(raw.clone().cast());
+            // We assume this can only happen in joined rooms, or something's very wrong.
+            new_rooms
+                .join
+                .entry(room_id.to_owned())
+                .or_insert_with(JoinedRoom::default)
+                .ephemeral
+                .push(raw.clone().cast());
         }
 
         for (room_id, raw) in &extensions.typing.rooms {
-            // Include the typing notifications in the room update, so the timeline is aware
-            // of those. We assume that those happen only in joined rooms.
-            let room_update =
-                new_rooms.join.entry(room_id.clone()).or_insert_with(JoinedRoom::default);
-            room_update.ephemeral.push(raw.clone().cast());
+            // We assume this can only happen in joined rooms, or something's very wrong.
+            new_rooms
+                .join
+                .entry(room_id.to_owned())
+                .or_insert_with(JoinedRoom::default)
+                .ephemeral
+                .push(raw.clone().cast());
+        }
+
+        // Rooms in `new_rooms.join` either have a timeline update, or a new read
+        // receipt. Update the read receipt accordingly.
+        for (room_id, joined_room_update) in &mut new_rooms.join {
+            if let Some(mut room_info) = changes
+                .room_infos
+                .get(room_id)
+                .cloned()
+                .or_else(|| self.get_room(room_id).map(|r| r.clone_info()))
+            {
+                // TODO only add the room if there was an update
+                compute_notifications(
```

Comment:
Should we check the room is encrypted before calling `compute_notifications`?

Reply:
Also, it's not clear to me where/when notifications are computed if the room isn't encrypted. Should it be documented somewhere around here?